### ISSUE TITLE
Everything is Working Great, quick question....

### ISSUE THREAD

**Issue by MushroomFleet** [cloneofsimo/lora#19]
as seen here, (example outputs included)
https://twitter.com/MushroomFleet/status/1602341447952437249
and below (just the merge experiments)

What are your advised Training settings? I did use a bunch that i used in other Dreambooth methods, but it would be interesting to know if you have a recommended setting for LORA because it is a little different. 

Thanks for your hard work in bringing this to us :)

![MichaelFassbender-LORA-AiNodes](https://user-images.githubusercontent.com/7318371/207101779-649c76e3-5e20-4227-81aa-9791deabc37a.png)

**Comment by cloneofsimo** [comment]
I really haven't experimented with LoRA to conclude anything, but for me,
using lr=1e-4, prior_preservation = False worked OK. here is more results from training 10 illustration images with 30000 steps. 
![image](https://user-images.githubusercontent.com/35953539/207116007-d8877471-bcfd-4d1f-899c-cebb185f579e.png)

**Comment by nitrosocke** [comment]
I haven't done any style training with it (yeah I know, how is this possible?..)
I used it mainly for subject training and refining a few celebrities in base SD, here are my results in conjunction with my dreambooth models:
![nitrosocke-lora-comp-01](https://user-images.githubusercontent.com/91015547/207127222-20e4d89a-1c14-491a-89dc-d36a29ac7bf3.jpg)
![nitrosocke-lora-comp-02](https://user-images.githubusercontent.com/91015547/207127234-a5cc615f-f217-4b3c-9a01-d70214731c6b.jpg)
![nitrosocke-lora-comp-03](https://user-images.githubusercontent.com/91015547/207127236-08ee0a5d-e8a1-4709-8e45-275bbc37be56.jpg)
![nitrosocke-lora-comp-04](https://user-images.githubusercontent.com/91015547/207127241-3a0df934-0d08-4bf8-b5c3-c65d11784669.jpg)
![nitrosocke-lora-comp-05](https://user-images.githubusercontent.com/91015547/207127244-f0f2fcee-5f79-4717-8852-a59b83842f29.jpg)

The settings I used where the following:
`--pretrained_model_name_or_path=$MODEL_NAME
  --weight_name=$WEIGHT_NAME
  --instance_data_dir=$INSTANCE_DIR
  --output_dir=$OUTPUT_DIR
  --instance_prompt="[celeb name]"
  --resolution=768
  --train_batch_size=4
  --mixed_precision="fp16"
  --use_8bit_adam
  --gradient_accumulation_steps=1
  --learning_rate=1e-4
  --lr_scheduler="constant"
  --save_steps=200
  --max_train_steps=1000`

**Comment by nitrosocke** [comment]
@pedrogengo Sure thing here is my workflow:
- Choose 5-10 images of a person
- Crop/resize to 768x768 for SD 2.1 training
- Following settings worked for me: _train_batch_size=4, mixed_precision="fp16", use_8bit_adam, learning_rate=1e-4, lr_scheduler="constant", save_steps=200, max_train_steps=1000_
- for systems with less VRAM than 24GB use a lower train_batch_size like 2 or 1.
- make sure to use xformers and 8bit adam
- for subjects already know to SD images*100 worked great, for subjects unknown to SD more steps or a higher LR are required
- training on a 3090 takes ~20 min for 1k steps

**Comment by nitrosocke** [comment]
@pedrogengo I did try training my wife and the settings didn't work. I haven't done any more testing on that but I assume it's something like 2k steps for 5 images. Would have to try though.

Prompt for Joseph was "modern disney style joseph gordon levitt", it's using my custom model and the JGL training .pt on top.

**Comment by MushroomFleet** [comment]
Thanks for all the responses everyone!! 

I generally only overwrite known concepts (like actors) with photos of myself. I'll take a look at all the settings you all posted and tweak mine accordingly :) 

I'm helping to test a local installed UI implementation that now included LORA dreambooth so any information is very useful for guiding users !!

**Comment by cloneofsimo** [comment]
Now you can also fine-tune CLIP encoder with LoRA as well, just like dreambooth option

Checkout fine-tuning with this shell-code:
https://github.com/cloneofsimo/lora/blob/master/run_lora_db_w_text.sh

Checkout using LoRAs this notebook (on example LoRA i've made): https://github.com/cloneofsimo/lora/blob/master/scripts/run_with_text_lora_also.ipynb

**Comment by amrrs** [comment]
@cloneofsimo Thanks for the work. Do you have any Colab / Notebook for basic training. I currently saw 4 scripts on the scripts folder, but not sure if you got the example there or somewhere else.

Basically, if i've got 4 768x768 images and I want to fine-tune SD2.0, then which is the script I should look at?

**Comment by cloneofsimo** [comment]
You can run:
https://github.com/cloneofsimo/lora/blob/master/run_lora_db_w_text.sh
with few parameters changed. defaults worked OK for me, and I even stopped at 2500 steps because it was overfitting.

**Comment by amrrs** [comment]
Thank you @cloneofsimo sorry for a dumb question. So the input images should be inside this folder which is assigned as the instance directory `INSTANCE_DIR="./data_example_text"` ?

**Comment by amerkay** [comment]
> @pedrogengo is it possible to run colab training with batch_size=5? It is the default setting but I'm getting CUDA_OUT_OF_MEMORY

Which GPU are you using? T4 15GB (standard) or A100 40GB (premium)? 

I'm trying to figure out ideal settings for the A100.

**Comment by cloneofsimo** [comment]
Nice work @pedrogengo! I'll update the readme.
Meanwhile it would be nice if there was also text encoder training. Would you like to update it?

**Comment by brian6091** [comment]
@pedrogengo @cloneofsimo the script train_lora_dreambooth.py seems to be missing a call for accelerate to manage the accumulation context:

https://huggingface.co/docs/accelerate/v0.13.2/en/package_reference/accelerator#accelerate.Accelerator.accumulate

**Comment by cloneofsimo** [comment]
Right gradient accumulation doesn't work now because it implicitly updates all other params wrapped inside it. So i removed it.

**Comment by cloneofsimo** [comment]
Yes I think so. But Im not really used to accelerate package so it was probably wasn't the way to fix it. I'll try to make it work with grad accumulation

**Comment by brian6091** [comment]
Just gonna drop a link to more training/tuning discussion here: 
https://github.com/cloneofsimo/lora/discussions/37

**Comment by scaraffe** [comment]
Is there a way to train new images with manually added captions?

**Comment by brian6091** [comment]
@scaraffe 

You can try this colab notebook:
https://github.com/brian6091/Dreambooth

that allows using captions with @cloneofsimo's lora training

**Comment by nayan-dhabarde** [comment]
> I haven't done any style training with it (yeah I know, how is this possible?..) I used it mainly for subject training and refining a few celebrities in base SD, here are my results in conjunction with my dreambooth models: ![nitrosocke-lora-comp-01](https://user-images.githubusercontent.com/91015547/207127222-20e4d89a-1c14-491a-89dc-d36a29ac7bf3.jpg) ![nitrosocke-lora-comp-04](https://user-images.githubusercontent.com/91015547/207127241-3a0df934-0d08-4bf8-b5c3-c65d11784669.jpg) ![nitrosocke-lora-comp-05](https://user-images.githubusercontent.com/91015547/207127244-f0f2fcee-5f79-4717-8852-a59b83842f29.jpg)
> 
> The settings I used where the following: `--pretrained_model_name_or_path=$MODEL_NAME --weight_name=$WEIGHT_NAME --instance_data_dir=$INSTANCE_DIR --output_dir=$OUTPUT_DIR --instance_prompt="[celeb name]" --resolution=768 --train_batch_size=4 --mixed_precision="fp16" --use_8bit_adam --gradient_accumulation_steps=1 --learning_rate=1e-4 --lr_scheduler="constant" --save_steps=200 --max_train_steps=1000`

What were lora hyper parameters and what are the number of input images you used for this
